The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13653752950/job/38168042135) [(plain)](https://github.com/rust-lang/rust/commit/8c9cd17179d5e2072eec41554161bd69b08de736/checks/38168042135/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04
ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
  make \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 2.816 Building wheels for collected packages: reuse
#12 2.817   Building wheel for reuse (pyproject.toml): started
#12 3.032   Building wheel for reuse (pyproject.toml): finished with status 'done'
#12 3.033   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=0c2fd2aaf7b0bf8d6e131220aff14712a774c2ca462f3204d25460cbcf610b63
#12 3.033   Stored in directory: /tmp/pip-ephem-wheel-cache-89deq2mt/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#12 3.036 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#12 3.445 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#12 3.445 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 3.993 Collecting virtualenv
#12 3.993 Collecting virtualenv
#12 4.078   Downloading virtualenv-20.29.2-py3-none-any.whl (4.3 MB)
#12 4.184      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 41.6 MB/s eta 0:00:00
#12 4.245 Collecting filelock<4,>=3.12.2
#12 4.260   Downloading filelock-3.17.0-py3-none-any.whl (16 kB)
#12 4.280 Collecting distlib<1,>=0.3.7
#12 4.294   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.302      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 92.4 MB/s eta 0:00:00
#12 4.335 Collecting platformdirs<5,>=3.9.1
#12 4.350   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#12 4.433 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.619 Successfully installed distlib-0.3.9 filelock-3.17.0 platformdirs-4.3.6 virtualenv-20.29.2
#12 DONE 4.7s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
DirectMap4k:      137152 kB
DirectMap2M:     8251392 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2025-02-18/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
fmt check
fmt: checked 5871 files
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/src/doc/rustc/src/platform-support/openharmony.md:32: trailing whitespace
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'venv'
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'virtualenv'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (25.0.1)
All checks passed!
checking python file formatting
27 files already formatted
checking C++ file formatting
checking C++ file formatting
some tidy checks failed
Command has failed. Rerun with -v to see more details.
  local time: Tue Mar  4 12:40:00 UTC 2025
  network time: Tue, 04 Mar 2025 12:40:00 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/ci/docker/scripts/ohos-sdk.sh
+++ b/src/ci/docker/scripts/ohos-sdk.sh
@@ -7,3 +7,4 @@ curl $URL | tar xz -C /tmp linux/native-linux-x64-5.0.0.71-Release.zip
 mkdir /opt/ohos-sdk
 cd /opt/ohos-sdk
 unzip -qq /tmp/linux/native-linux-x64-5.0.0.71-Release.zip
+rm /tmp/linux/native-linux-x64-5.0.0.71-Release.zip

--- a/src/ci/github-actions/jobs.yml
+++ b/src/ci/github-actions/jobs.yml
@@ -176,7 +176,7 @@ auto:
     <<: *job-linux-4c
 
   - name: dist-ohos
-    <<: *job-linux-4c
+    <<: *job-linux-4c-largedisk
 
   - name: dist-powerpc-linux
     <<: *job-linux-4c

--- a/src/ci/run.sh
+++ b/src/ci/run.sh
@@ -113,7 +113,11 @@ export RUST_RELEASE_CHANNEL=$(releaseChannel)
 RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --release-channel=$RUST_RELEASE_CHANNEL"
 
 if [ "$DEPLOY$DEPLOY_ALT" = "1" ]; then
-  RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --enable-llvm-static-stdcpp"
+  if [[ "$CI_JOB_NAME" == *ohos* ]]; then
+    RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --disable-llvm-static-stdcpp"
+  else
+    RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --enable-llvm-static-stdcpp"
+  fi
   RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --set rust.remap-debuginfo"
 
   if [ "$DEPLOY_ALT" != "" ] && isLinux; then

--- a/src/doc/rustc/src/platform-support/openharmony.md
+++ b/src/doc/rustc/src/platform-support/openharmony.md
@@ -29,6 +29,9 @@ The targets require a reasonably up-to-date OpenHarmony SDK on the host.
 
 The targets support `cargo`, which require [ohos-openssl](https://github.com/ohos-rs/ohos-openssl).
 
+`miri` isn't supported yet, since its dependencies (`libffi` and `tikv-jemalloc-sys`) don't support
+compiling for the OHOS targets.
+
 ## Setup
 
 The OpenHarmony SDK doesn't currently support Rust compilation directly, so

